core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -305,18 +292,18 @@ private Object primaryReturnHandler(InvocationContext ctx, VisitableCommand visi
          if (trace) log.tracef("Skipping the replication of the conditional command as it did not succeed on primary owner (%s).", command);
          return localResult;
       }
-      // check if a single owner has been configured and the target for the key is the local address
-      boolean isSingleOwnerAndLocal = cacheConfiguration.clustering().hash().numOwners() == 1;
-      if (isSingleOwnerAndLocal) {
+      DistributionInfo distributionInfo = checkTopologyId(command).getDistribution(command.getKey());
+      Collection<Address> owners = distributionInfo.writeOwners();
+      if (owners.size() == 1) {
+         // There are no backups, skip the replication part.
          return localResult;
       }
-      ConsistentHash writeCH = checkTopologyId(command).getWriteConsistentHash();
-      List<Address> recipients = writeCH.isReplicated() ? null : writeCH.locateOwners(command.getKey());
 
       // Cache the matcher and reset it if we get OOTE (or any other exception) from backup
       ValueMatcher originalMatcher = command.getValueMatcher();
       // Ignore the previous value on the backup owners
       command.setValueMatcher(ValueMatcher.MATCH_ALWAYS);
+      Collection<Address> recipients = isReplicated ? null : owners;
```

Comment:
Asymmetric caches is exactly why I wanted to use `null` here, to keep the RPC using broadcast while a node joins instead of switching between broadcast and unicast. For performance it probably doesn't make a big difference, but the UNICAST3/NAKACK2 switch also hurts consistency in async non-tx caches.